File: 정민지/SEA_1206.py
import sys
input = sys.stdin.readline

for tc in range(1):
    n = int(input())
    lst = list(map(int, input().split()))

    result = 0
    for i in range(2, len(lst)-2):
        val = lst[i] - max(lst[i-2], lst[i-1], lst[i+1], lst[i+2])
        if val > 0:
            result += val

    print(f'#{tc+1} {result}')

File: 정민지/java/src/sea/SEA_1206.java
package sea;

import java.io.*;
import java.util.*;

public class SEA_1206 {

	public static void main(String args[]) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

		for (int tc = 1; tc <= 10; tc++) {
			int n = Integer.parseInt(br.readLine());
			int[] arr = Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();
			int result = 0;

			for (int i = 2; i < n - 2; i++) {
				int maxVal = Math.max(Math.max(arr[i+2], arr[i+1]), Math.max(arr[i-2], arr[i-1]));

				if (maxVal < arr[i]) {
					result += arr[i] - maxVal;
				}
			}

			bw.write("#" + tc + " " + result);
			bw.newLine();
		}

		bw.flush();
		bw.close();
	}
}
